.github/workflows/dotnet-build.yml

```diff
@@ -268,3 +321,52 @@ jobs:
           --source https://api.nuget.org/v3/index.json
           --api-key ${{ secrets.NUGET_PACKAGE_PUSH_TOKEN }}
           --skip-duplicate
+
+  run_benchmarks:
```

Comment:
Can we run benchmarks as part of the main pipeline? I want the pipeline runs as fast as possible. Also do we need benchmarks on every pipeline? Maybe it can be a manual step?

Reply:
Good point! I've edited [our Ruleset](https://github.com/CommunityToolkit/Maui/settings/rules/2541778) so that the Benchmark workflow is not required to merge the PR. 

<img width="607" alt="Screenshot 2025-01-19 at 8 22 38 AM" src="https://github.com/user-attachments/assets/86f29af2-62bc-4930-9999-feb8f6856224" />

They'll still run, giving us benchmark data that we can leverage for Analyzer updates, but we don't have to wait on them to finish to merge.